element_calcium_imaging/field_processing.py

```diff
@@ -293,18 +293,21 @@ def make(self, key):
             ops_path = find_full_path(processed_root_data_dir, extra_params["ops_path"])
 
             # calculate aspect ratio and diameter
-            um_height, um_width = (scan.ScanInfo.Field & key).fetch1(
-                "um_height", "um_width"
-            )
-            aspect = um_height / um_width
+            px_height, px_width, um_height, um_width = (
+                scan.ScanInfo.Field & key
+            ).fetch1("px_height", "px_width", "um_height", "um_width")
+            aspect = (um_height / px_height) / (um_width / px_width)
             params["aspect"] = aspect
 
-            diameter = params.get(
-                "diameter", 10
-            )  # default diameter is 10um unless specified in `ProcessingParamSet`
+            diameter = params.get("diameter")  # get diameter
+            if diameter == 0:
```

Comment:
In which case would `diameter` set in `params` ever be 0? And if it is intentionally set to `0` by users, perhaps we should not modify here